File: MiniDOMBase/DMD/Anagrafica/Tasks/RegoleTaskLavorazione/RegolaTaskLavorazioneCursor.vb
Imports minidom
Imports minidom.Sistema
Imports minidom.Databases



Partial Public Class Anagrafica

    <Serializable>
    Public Class RegolaTaskLavorazioneCursor
        Inherits DBObjectCursor(Of RegolaTaskLavorazione)

        Private m_Nome As New CCursorFieldObj(Of String)("Nome")
        Private m_Descrizione As New CCursorFieldObj(Of String)("Descrizione")
        Private m_IDStatoSorgente As New CCursorField(Of Integer)("IDStatoSorgente")
        Private m_IDStatoDestinazione As New CCursorField(Of Integer)("IDStatoDestinazione")
        Private m_Flags As New CCursorField(Of Integer)("Flags")
        Private m_NomeHandler As New CCursorFieldObj(Of String)("NomeHandler")
        Private m_ContextType As New CCursorFieldObj(Of String)("ContextType")
        Private m_Ordine As New CCursorField(Of Integer)("Ordine")

        Public Sub New()
        End Sub

        Public ReadOnly Property ContextType As CCursorFieldObj(Of String)
            Get
                Return Me.m_ContextType
            End Get
        End Property

        Public ReadOnly Property Nome As CCursorFieldObj(Of String)
            Get
                Return Me.m_Nome
            End Get
        End Property

        Public ReadOnly Property NomeHandler As CCursorFieldObj(Of String)
            Get
                Return Me.m_NomeHandler
            End Get
        End Property

        Public ReadOnly Property Descrizione As CCursorFieldObj(Of String)
            Get
                Return Me.m_Descrizione
            End Get
        End Property

        Public ReadOnly Property IDStatoSorgente As CCursorField(Of Integer)
            Get
                Return Me.m_IDStatoSorgente
            End Get
        End Property

        Public ReadOnly Property IDStatoDestinazione As CCursorField(Of Integer)
            Get
                Return Me.m_IDStatoDestinazione
            End Get
        End Property

        Public ReadOnly Property Ordine As CCursorField(Of Integer)
            Get
                Return Me.m_Ordine
            End Get
        End Property

        Public ReadOnly Property Flags As CCursorField(Of Integer)
            Get
                Return Me.m_Flags
            End Get
        End Property

        Public Overrides Function GetTableName() As String
            Return "tbl_TaskLavorazioneRegole"
        End Function


        Protected Friend Overrides Function GetConnection() As Databases.CDBConnection
            Return Anagrafica.TasksDiLavorazione.Database
        End Function

        Protected Overrides Function GetModule() As CModule
            Return Nothing ' Anagrafica.RegoleTasksLavorazione.Module
        End Function
    End Class


End Class

File: MiniDOMBase/DMD/Anagrafica/Tasks/RegoleTaskLavorazione/RegolaTaskLavorazioneCursor.cs
using System;
using System.Collections;
using DMD;
using DMD.XML;
using DMD.Databases;
using DMD.Databases.Collections;
using minidom.repositories;
using static minidom.Anagrafica;
using static minidom.Sistema;

namespace minidom
{
    public partial class Anagrafica
    {

        /// <summary>
        /// Cursore sulla tabella delle regole di lavorazione
        /// </summary>
        [Serializable]
        public class RegolaTaskLavorazioneCursor
            : minidom.Databases.DBObjectCursor<RegolaTaskLavorazione>
        {
            private DBCursorStringField m_Nome = new DBCursorStringField("Nome");
            private DBCursorStringField m_Descrizione = new DBCursorStringField("Descrizione");
            private DBCursorField<int> m_IDStatoSorgente = new DBCursorField<int>("IDStatoSorgente");
            private DBCursorField<int> m_IDStatoDestinazione = new DBCursorField<int>("IDStatoDestinazione");
            private DBCursorField<int> m_Flags = new DBCursorField<int>("Flags");
            private DBCursorStringField m_NomeHandler = new DBCursorStringField("NomeHandler");
            private DBCursorStringField m_ContextType = new DBCursorStringField("ContextType");
            private DBCursorField<int> m_Ordine = new DBCursorField<int>("Ordine");

            /// <summary>
            /// Costruttore
            /// </summary>
            public RegolaTaskLavorazioneCursor()
            {
            }

            /// <summary>
            /// ContextType
            /// </summary>
            public DBCursorStringField ContextType
            {
                get
                {
                    return m_ContextType;
                }
            }

            /// <summary>
            /// Nome
            /// </summary>
            public DBCursorStringField Nome
            {
                get
                {
                    return m_Nome;
                }
            }

            /// <summary>
            /// NomeHandler
            /// </summary>
            public DBCursorStringField NomeHandler
            {
                get
                {
                    return m_NomeHandler;
                }
            }

            /// <summary>
            /// Descrizione
            /// </summary>
            public DBCursorStringField Descrizione
            {
                get
                {
                    return m_Descrizione;
                }
            }

            /// <summary>
            /// IDStatoSorgente
            /// </summary>
            public DBCursorField<int> IDStatoSorgente
            {
                get
                {
                    return m_IDStatoSorgente;
                }
            }

            /// <summary>
            /// IDStatoDestinazione
            /// </summary>
            public DBCursorField<int> IDStatoDestinazione
            {
                get
                {
                    return m_IDStatoDestinazione;
                }
            }

            /// <summary>
            /// Ordine
            /// </summary>
            public DBCursorField<int> Ordine
            {
                get
                {
                    return m_Ordine;
                }
            }

            /// <summary>
            /// Flags
            /// </summary>
            public DBCursorField<int> Flags
            {
                get
                {
                    return m_Flags;
                }
            }

            //public override string GetTableName()
            //{
            //    return "tbl_TaskLavorazioneRegole";
            //}

            //protected internal override CDBConnection GetConnection()
            //{
            //    return TasksDiLavorazione.Database;
            //}


            /// <summary>
            /// Repository
            /// </summary>
            /// <returns></returns>
            protected override CModulesClass GetModule()
            {
                return minidom.Anagrafica.RegoleTasksLavorazione; //.Module
            }
        }
    }
}